conans/test/unittests/tools/cmake/test_cmakedeps.py

```diff
@@ -163,3 +163,41 @@ def test_component_name_same_package():
         data_cmake = files["mypkg-release-x86-data.cmake"]
         assert 'set(mypkg_mypkg_mypkg_INCLUDE_DIRS_RELEASE ' \
                '"${mypkg_PACKAGE_FOLDER_RELEASE}/includedirs1")' in data_cmake
+
+
+def test_component_cmake_deps_links_flags():
+    # https://github.com/conan-io/conan/issues/8703
+    conanfile = ConanFile(Mock(), None)
+    conanfile._conan_node = Mock()
+    conanfile._conan_node.context = "host"
+    conanfile.settings = "os", "compiler", "build_type", "arch"
+    conanfile.initialize(Settings({"os": ["Windows"],
+                                   "compiler": ["gcc"],
+                                   "build_type": ["Release"],
+                                   "arch": ["x86"]}), EnvValues())
+    conanfile.settings.build_type = "Release"
+    conanfile.settings.arch = "x86"
+
+    cpp_info = CppInfo("mypkg", "dummy_root_folder1")
+    # https://github.com/conan-io/conan/issues/8811 regression, fix with explicit - instead of /
+    cpp_info.components['cmp1'].sharedlinkflags = ["-NODEFAULTLIB", "-OTHERFLAG"]
+    cpp_info.components['cmp1'].exelinkflags = ["-OPT:NOICF", "-s FULL_ES3=1"]
+    cpp_info.components['cmp1'].objects = ["myobject.o"]
+    conanfile_dep = ConanFile(Mock(), None)
+    conanfile_dep.cpp_info = cpp_info
+    conanfile_dep._conan_node = Mock()
+    conanfile_dep._conan_node.ref = ConanFileReference.loads("mypkg/1.0")
+    conanfile_dep._conan_node.context = "host"
+    conanfile_dep.folders.set_base_package("/path/to/folder_dep")
+
+    with mock.patch('conans.ConanFile.dependencies', new_callable=mock.PropertyMock) as mock_deps:
+        req = Requirement(ConanFileReference.loads("OriginalDepName/1.0"))
+        mock_deps.return_value = ConanFileDependencies({req: ConanFileInterface(conanfile_dep)})
+
+        cmakedeps = CMakeDeps(conanfile)
+        files = cmakedeps.content
+        data_cmake = files["mypkg-release-x86-data.cmake"]
+        assert 'set(mypkg_mypkg_cmp1_SHARED_LINK_FLAGS_RELEASE "-NODEFAULTLIB;-OTHERFLAG")' in data_cmake
+        assert 'set(mypkg_mypkg_cmp1_EXE_LINK_FLAGS_RELEASE "-OPT:NOICF;-s FULL_ES3=1")' in data_cmake
+        assert 'set(mypkg_mypkg_cmp1_OBJECTS_RELEASE "${mypkg_PACKAGE_FOLDER_RELEASE}/myobject.o")' \
+               in data_cmake
```

Comment:
This test is duplicated. Have a look at this PR (merged) https://github.com/conan-io/conan/pull/9980/files